pyproject.toml

```diff
@@ -39,36 +39,32 @@ cu12 = [
     "nvidia-nvjitlink-cu12"
 ]
 test = [
+    "pre-commit",
     "psutil",
     "cffi",
     "pytest",
     "pytest-xdist",
     "filecheck",
 ]
 test-cu11 = [
-    "numba-cuda[test]",
+    "pre-commit",
+    "psutil",
+    "cffi",
+    "pytest",
+    "pytest-xdist",
+    "filecheck",
     "nvidia-curand-cu11",
 ]
 test-cu12 = [
```

Comment:
I think we're missing `pre-commit` from here and is another reason to use `numba-cuda[test]` instead of duplicating all of the dependencies

Reply:
That is what I had written, and Graham asked that I not use self referential dependency groups. @gmarkall do you have any other thoughts about this? Are we concerned about self-referential dependency groups if other packages in our ecosystem are using them?